agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -89,13 +95,15 @@ type TaskEngineState interface {
 type DockerTaskEngineState struct {
 	lock sync.RWMutex
 
-	tasks          map[string]*apitask.Task                            // taskarn -> apitask.Task
-	idToTask       map[string]string                                   // DockerId -> taskarn
-	taskToID       map[string]map[string]*apicontainer.DockerContainer // taskarn -> (containername -> c.DockerContainer)
-	idToContainer  map[string]*apicontainer.DockerContainer            // DockerId -> c.DockerContainer
-	eniAttachments map[string]*apieni.ENIAttachment                    // ENIMac -> apieni.ENIAttachment
-	imageStates    map[string]*image.ImageState
-	ipToTask       map[string]string // ip address -> task arn
+	tasks                       map[string]*apitask.Task                            // taskarn -> apitask.Task
+	idToTask                    map[string]string                                   // DockerId -> taskarn
+	taskToID                    map[string]map[string]*apicontainer.DockerContainer // taskarn -> (containername -> c.DockerContainer)
+	idToContainer               map[string]*apicontainer.DockerContainer            // DockerId -> c.DockerContainer
+	eniAttachments              map[string]*apieni.ENIAttachment                    // ENIMac -> apieni.ENIAttachment
+	imageStates                 map[string]*image.ImageState
+	ipToTask                    map[string]string // ip address -> task arn
+	v3EndpointIDToTask          map[string]string // container's v3 endpoint id -> taskarn
+	v3EndpointIDToContainerName map[string]string // container's v3 endpoint id -> container name
```

Comment:
Can this be map of id -> container Docker ID, as we already have a map of Docker ID to containers?